packages/react/src/components/Forms/Select/mono-searchable.tsx

```diff
@@ -43,21 +43,36 @@ export const SelectMonoSearchable = (props: SubProps) => {
 
   // When component is controlled, this useEffect will update the local selected item.
   useEffect(() => {
-    if (props.downshiftProps.initialSelectedItem !== undefined) {
+    if (inputFilter) {
       return;
     }
+
+    const selectedItem = downshiftReturn.selectedItem
+      ? optionToValue(downshiftReturn.selectedItem)
+      : undefined;
+
+    if (selectedItem === props.value) {
+      return;
+    }
+
     const optionToSelect = props.options.find(
       (option) => optionToValue(option) === props.value,
     );
     downshiftReturn.selectItem(optionToSelect ?? null);
-  }, [props.value, props.options, props.downshiftProps]);
+  }, [props.value, props.options, inputFilter]);
 
   // Even there is already a value selected, when opening the combobox menu we want to display all available choices.
   useEffect(() => {
     if (downshiftReturn.isOpen) {
-      setOptionsToDisplay(props.options);
+      setOptionsToDisplay(
+        inputFilter
+          ? props.options.filter(getOptionsFilter(inputFilter))
+          : props.options,
+      );
+    } else {
+      setInputFilter(undefined);
```

Comment:
Nice way of refactoring the code in order to include all the deps, I really like the twist :)